Application/Education/ClassRegister/Diary/Frontend.php

```diff
@@ -0,0 +1,859 @@
+                    $tblGroup ? $tblGroup->getId() : null
+                ));
+        }
+
+        $columns = array();
+        if ($tblDivision) {
+            $tblPersonList = Division::useService()->getStudentAllByDivision($tblDivision);
+        } elseif ($tblGroup) {
+            if (($tblPersonList = Group::useService()->getPersonAllByGroup($tblGroup))) {
+                $tblPersonList = $this->getSorter($tblPersonList)->sortObjectBy('LastFirstName');
+            }
+
+        } else {
+            $tblPersonList = false;
+        }
+        if ($tblPersonList) {
+            foreach ($tblPersonList as $tblPerson) {
+                $columns[$tblPerson->getId()] = new FormColumn(new CheckBox('Data[Students][' . $tblPerson->getId() . ']',
+                    $tblPerson->getLastFirstName(), 1), 4);
+            }
+        }
+        // deaktivierte ausgewählte Schüler hinzufügen
+        if (!empty($setStudents)) {
+            foreach ($setStudents as $personId => $tblStudent) {
+                if (!isset($columns[$personId])) {
+                    $columns[$tblStudent->getId()] = new FormColumn(new CheckBox('Data[Students][' . $tblStudent->getId() . ']',
+                        new Strikethrough($tblStudent->getLastFirstName()), 1), 4);
+                }
+            }
+        }
+
+        return (new Form(
+            new FormGroup(array(
+                new FormRow(array(
+                    new FormColumn(
+                        (new DatePicker('Data[Date]', '', 'Datum', new Calendar()))->setRequired()
+                    , 6),
+                    new FormColumn(
+                        new TextField('Data[Place]', '', 'Ort')
+                    , 6),
+                )),
+                new FormRow(array(
+                    new FormColumn(
+                        new TextField('Data[Subject]', 'Titel', 'Titel', new Calendar())
+                    ),
+                )),
+                new FormRow(array(
+                   new FormColumn(array(
+                       new Warning('Wenn kein Schüler ausgewählt wird, handelt es sich um einen Klasseneintrag.')
+                   ))
+                )),
+                new FormRow(
+                    $columns
+                ),
+                new FormRow(array(
+                    new FormColumn(
+                        new TextArea('Data[Content]', 'Bemerkungen', 'Bemerkungen', new Edit())
+                    ),
+                )),
+                new FormRow(array(
+                    new FormColumn(
+                        $saveButton
+                    )
+                )),
+            ))
+        ))->disableSubmitAction();
+    }
+
+    /**
+     * @param TblDivision|null $tblDivision
+     * @param TblGroup|null $tblGroup
+     * @param TblPerson $tblStudentPerson
+     *
+     * @return TableData
+     */
+    public function loadDiaryTable(TblDivision $tblDivision = null, TblGroup $tblGroup = null, TblPerson $tblStudentPerson = null)
```

Comment:
Codacy found an issue: [The method loadDiaryTable() has a Cyclomatic Complexity of 21. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)